src/Domain/SizeOfElement.hpp

```diff
@@ -0,0 +1,58 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines function size_of_element
+
+#pragma once
+
+#include <cstddef>
+#include <string>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class DataVector;
+template <size_t VolumeDim>
+class Mesh;
+namespace Tags {
+template <size_t Dim, typename Frame>
+struct Coordinates;
+template <size_t VolumeDim>
+struct Mesh;
+}  // namespace Tags
+/// \endcond
+
+/*!
+ * \ingroup ComputationalDomainGroup
+ * \brief Compute the inertial-coordinate size of an element along each of its
+ * logical directions.
+ *
+ * For each logical direction, compute the mean position (in inertial
+ * coordinates) of the element's lower and upper faces in that direction.
+ * This is done by simply averaging the coordinates of the face grid points.
+ * The size of the element along this logical direction is then the distance
+ * between the mean positions of the lower and upper faces.
+ * Note that for curved elements, this is an approximate measurement of size.
+ */
+template <size_t VolumeDim>
+tnsr::I<double, VolumeDim> size_of_element(
```

Comment:
Discussion with @wthrowe, @nilsdeppe, @kidder (and others) concluded that this function should return a different type, at least a `tnsr::I<double, VolumeDim, Frame::Logical>` but probably better a `std::array<double, VolumeDim>`.